collective/honeypot/utils.py

```diff
@@ -73,6 +81,9 @@ def get_form(request):
         form = request.form
     else:
         form = request
+    if not form and getattr(request, "CONTENT_TYPE", "") == "application/json":
```

Comment:
Have you tried this? I would expect in the case of `form = request` that `not form` will always be false.
But maybe that is almost never the case.

Reply:
probably you're right..i never had the case where request don't have form attribute i think.
With a restapi call it's an empty dict.

btw i tried to refactor this piece of code, let me know if it's better